dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityQueryParams.java

```diff
@@ -614,6 +614,9 @@ public TrackedEntityQueryParams filterBy(TrackedEntityAttribute tea, QueryFilter
 
   public TrackedEntityQueryParams filterBy(TrackedEntityAttribute tea) {
     this.filters.putIfAbsent(tea, Collections.singletonList(null));
+    if (this.filters.get(tea).isEmpty()) {
```

Comment:
I think this warrants a comment as we won't remember 😅 

Also I think we need to document the special role of the singleton `null` somewhere. You describe it in the PR but maybe it should be part of the javadoc for the `filters` field.